chapter.md

```diff
@@ -0,0 +1,735 @@
+> instead, it's the blank slate that would be updated from a database
+> as the application is being bootstrapped.
+> And yes,
+> that bootstrap process would be implemented as
+> a series of update actions.
+
+Another early decision is that we will define functions to create actions,
+and that each action will have two elements:
+a string called `type` (which Redux requires),
+and a sub-object called `payload` that will contain
+the path to the part of the store it applies to
+and whatever extra values are needed to update the state.
+Redux doesn't mandate this,
+but again,
+experience teaches us that creating literal objects
+in many different places throughout our program
+is just as risky as using magic numbers
+rather than defining a constant and referring to it.
+
+As an example of an action creation function,
+here's one that saves a form's value:
+
+```ts
+const saveForm = (path, value) => ({
+  type: 'SAVE_FORM',
+  payload: {
+    path: path,
+    value: value
+  }
+});
+```
+
+Now that we know the shapes of our store and our action,
+the first version of the reducer is easy to write.
+To keep it short,
+we rely on three helper functions from [Ramda][ramda-site],
+a library of functional utilities for JavaScript:
+
+*   `assocPath`: makes a shallow clone of an object,
+    replacing the specified property with the specified value as it does so.
+    (Think of this as "make me a copy of X, but with Y set to Z".)
+*   `lensPath`: returns a *lens* that points to a particular location in a nested data structure.
+    (This is like a path in a filesystem, and is used for similar purposes.)
+*   `merge`: create a copy of one object with properties merged in from a second object.
+    (This is like `assocPath`, but using a second object to get multiple changes at once.)
+
+With these helpers in hand,
+our `formReducer` looks like this:
+
+```ts
+import { lensPath, assocPath, merge } from 'ramda';
+
+export function formReducer(state = initialState: IForm, action) {
+  switch (action.type) {
+
+  case 'SAVE_FORM':
+    const lensForProp = lensPath(action.payload.path);
+    return assocPath(
+      action.payload.path,
+      merge(view(lensForProp, state), action.payload.value),
+      state
+    );
+
+  default:
+    return state;
+  }
+}
+```
+
+There is actually less going on here than first appears.
+Working from the outside in,
+if the action's type is *not* `SAVE_FORM`,
+then `formReducer` returns the original state without any changes.
+This means that we can safely chain this reducer together
+with others that handle other parts of our user interface:
+as long as they all use distinct keys for their actions,
+they will watch the state fly by without doing anything we don't want them to.
+
+If the action does have the right type,
+`formReducer` uses `lensPath` to figure out which part of the state needs to be updated,
+then uses `merge` and `assocPath` to create a copy of the state
```

Comment:
what do you think in mentioning here again `... to create a shallow copy of the state`?